lib/cdo/mailjet.rb

```diff
@@ -0,0 +1,110 @@
+require 'mailjet'
+require 'honeybadger/ruby'
+require_relative './shared_constants/mailjet_constants'
+
+module MailJet
+  include MailJetConstants
+
+  API_KEY = CDO.try(:mailjet_api_key).freeze
+  SECRET_KEY = CDO.try(:mailjet_secret_key).freeze
+
+  def self.enabled?
+    return false unless DCDO.get('use_mailjet', false)
+    !API_KEY.nil? && !SECRET_KEY.nil?
+  end
+
+  def self.subaccount
+    case Rails.env
+    when 'production'
+      'production'
+    when 'staging'
+      'staging'
+    else
+      'development'
+    end
+  end
+
+  def self.create_contact_and_send_welcome_email(user)
+    return unless enabled?
+
+    return if user&.id.nil?
+    return unless user.teacher?
+
+    create_contact(user.email, user.name, user.created_at.to_datetime.rfc3339)
+    send_template_email(
+      user.email,
+      EMAILS[:welcome][:from_address],
+      user.name,
+      EMAILS[:welcome][:from_name],
+      EMAILS[:welcome][:template_id][subaccount.to_sym]
+    )
+  end
+
+  def self.create_contact(email, name, sign_up_date)
+    return unless enabled?
+
+    Mailjet.configure do |config|
+      config.api_key = API_KEY
+      config.secret_key = SECRET_KEY
+      config.api_version = "v3"
+    end
+
+    # If a contact already exists, Mailjet will raise an error.
+    # This shouldn't happen as user emails should be unique, but
+    # we don't want to block sign up if it does.
+    begin
+      Mailjet::Contact.create(
+        is_excluded_from_campaigns: true,
+        email: email,
+        name: name
+      )
+    rescue => exception
+      Honeybadger.notify(exception)
+    end
+
+    # Most likely, the above would fail if a contact already exists.
+    # In that case, we want to update the contact with the sign up date.
```

Comment:
Is this in the case that (for example), someone would opt in to receive the 'host an hour of code' email but not yet have an account?

Reply:
oo true, I always forget about how many ways we get user emails. That seems like a likely pathway!